src/useradd.c

```diff
@@ -749,6 +750,9 @@ static int set_defaults (void)
 	if (prefix[0]) {
 		free(default_file);
 	}
+	if (ofp != NULL) {
+		fclose (ofp);
+	}
```

Comment:
I'm still not convinced by this patch.  I've opened an alternative here: <https://github.com/shadow-maint/shadow/pull/997>

Reply:
I prefer your proposal so I'm dropping the patch. Thanks for the help!